server/src/main/java/org/elasticsearch/inference/InferenceProvider.java

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.inference;
+
+import org.elasticsearch.action.ActionListener;
+
+import java.util.List;
+
+/**
+ * Provides NLP text inference results. Plugins can implement this interface to provide their own inference results.
+ */
+public interface InferenceProvider {
```

Comment:
New interface for doing text inference from server, without referencing InferenceAction explicitly. This will be provided by a new InferencePlugin class.

Reply:
@benwtrent , @jimczi , what do you think of this approach? This class is used as an abstraction on server that gets filled in by the Inference Plugin

Reply:
I am fine with this as long as we prevent other plugins to override.

Reply:
Is there an effective way of preventing other plugins to override this? I've checked that there's a single provider, but I can't think of a way in which we can prevent user defined plugins to implement this if they exclude the Inference plugin....

Reply:
I think that's fine, it won't be documented so if users disable the inference plugin they're on uncharted territory ;)